lab_1_classify_by_unigrams/main.py

```diff
@@ -6,106 +6,193 @@
-    """
-    Calculate mean squared error between predicted and actual values.
-
-    Args:
-        predicted (list): A list of predicted values
-        actual (list): A list of actual values
-
-    Returns:
-        float | None: The score
-
-    In case of corrupt input arguments, None is returned
-    """
-
-
-def compare_profiles(
-    unknown_profile: dict[str, str | dict[str, float]],
-    profile_to_compare: dict[str, str | dict[str, float]],
-) -> float | None:
-    """
-    Compare profiles and calculate the distance using symbols.
-
-    Args:
-        unknown_profile (dict[str, str | dict[str, float]]): A dictionary of an unknown profile
-        profile_to_compare (dict[str, str | dict[str, float]]): A dictionary of a profile
-            to compare the unknown profile to
-
-    Returns:
-        float | None: The distance between the profiles
-
-    In case of corrupt input arguments or lack of keys 'name' and
-    'freq' in arguments, None is returned
-    """
-
-
-def detect_language(
-    unknown_profile: dict[str, str | dict[str, float]],
-    profile_1: dict[str, str | dict[str, float]],
-    profile_2: dict[str, str | dict[str, float]],
-) -> str | None:
-    """
-    Detect the language of an unknown profile.
-
-    Args:
-        unknown_profile (dict[str, str | dict[str, float]]): A dictionary of a profile
-            to determine the language of
-        profile_1 (dict[str, str | dict[str, float]]): A dictionary of a known profile
-        profile_2 (dict[str, str | dict[str, float]]): A dictionary of a known profile
-
-    Returns:
-        str | None: A language
-
-    In case of corrupt input arguments, None is returned
-    """
-
+def tokenize(text):
+    if not isinstance(text, str):
+        return None
+    extra = ',.:?!;%$#№"&()[]{}~’º01‘23456789 -\n><*=@'
+    text_only_letters = ''
+    for symbol in text:
+        if symbol not in extra:
+            text_only_letters += symbol
+        else:
+            continue
+    text_only_letters = list(text_only_letters.lower())
+    #print(text_only_letters)
+    return text_only_letters
+
+
+def calculate_frequencies(tokens):
+    if not isinstance(tokens, list):
+        return None
+    if isinstance(tokens, list):
+        for element in tokens:
+            if not isinstance(element, str):
+                return None
+    dictionary = {}
+    token_number = len(tokens)
+    for letter in tokens:
+        if letter not in dictionary.keys():
+            dictionary[letter] = 1/token_number
+        else:
+            dictionary[letter] = (dictionary[letter]*token_number + 1)/token_number
+    #print(dictionary)
+    return dictionary
+
+
+def create_language_profile(language, text):
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    dictionary = calculate_frequencies(tokenize(text))
+    language_profile = {}
+    language_profile['name'] = language
+    language_profile['freq'] = dictionary
```

Comment:
concatenate these strings and immediately create a dictionary

Reply:
done